src/android/jni/debug_protocol.cpp

```diff
@@ -2,152 +2,61 @@
 // Copyright(c) 2019 Intel Corporation. All Rights Reserved.
 
 #include <jni.h>
-#include <memory>
-#include <vector>
 #include "error.h"
-#include <iostream>
 #include <fstream>
-#include <string>
+#include "../../../include/librealsense2/h/rs_internal.h"
+#include "../../../common/parser.hpp"   //needed for auto completion in nGetCommands
 
-#include "../../../include/librealsense2/rs.h"
-#include "../../../include/librealsense2/hpp/rs_device.hpp"
-#include "../../api.h"
-#include "../../../common/parser.hpp"
 
-using std::string;
-using std::hex;
-
-std::vector<uint8_t> send_and_receive_raw_data(JNIEnv *env, rs2_device * dev, const std::vector<uint8_t>& input)
-{
-    std::vector<uint8_t> results;
+extern "C"
+JNIEXPORT jbyteArray JNICALL
+Java_com_intel_realsense_librealsense_DebugProtocol_nSendAndReceiveRawData(JNIEnv *env, jclass type,
+                                                                           jlong handle,
+                                                                           jbyteArray buffer_) {
+    jbyte *buffer = env->GetByteArrayElements(buffer_, NULL);
+    jsize length = env->GetArrayLength(buffer_);
 
-    rs2_error *e = NULL;
-    std::shared_ptr<const rs2_raw_data_buffer> list(
-            rs2_send_and_receive_raw_data(dev, (void*)input.data(), (uint32_t)input.size(), &e),
-            rs2_delete_raw_data);
+    rs2_error* e = NULL;
+    const rs2_raw_data_buffer* response_bytes =
```

Comment:
The native buffers should be wrapped with shared_ptr for RAII

Reply:
This is C code.

Reply:
understood - done